paddle/fluid/imperative/reducer.cc

```diff
@@ -527,6 +527,7 @@ void Reducer::TraverseBackwardGraph(
 void Reducer::PrepareForBackward(
     const std::vector<std::shared_ptr<imperative::VarBase>> &outputs) {
   VLOG(3) << "after forward, then reset count for backward.";
+  grad_need_hooks_ = true;
```

Comment:
Add a note to explain the role of this parameter

Reply:
Thanks. Already added notes in Line 212~215 of paddle/fluid/imperative/reducer.h